resources/assets/scripts/app/controllers/track.coffee

```diff
@@ -72,4 +72,19 @@ angular.module('ponyfm').controller "track", [
 
             playlists.addTrackToPlaylist(playlist.id, $scope.track.id).done (res) ->
                 playlist.message = res.message
+
+        $scope.getCachedTrack = (id, format) ->
+          $scope.isEncoding = true
+
+          cachedTrack.download('tracks', id, format).then (response) ->
+            trackUrl = response
```

Comment:
Can we get away with using only `$scope.trackUrl` here instead of two very similar variables?